pkg/operator/controller/controller.go

```diff
@@ -61,7 +63,13 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 
 // Config holds all the things necessary for the controller to run.
 type Config struct {
+	// Client is the kube Client and it will refresh scheme/mapper fields if needed
+	// to detect some resources like ServiceMonitor which could get registered after
+	// the client creation.
+	// Since this controller is running in single threaded mode,
+	// we do not need to synchronize when changing rest scheme/mapper fields.
 	Client                 client.Client
```

Comment:
Do we need to pass client anymore? I think accepting one implies you can share a given one, but we're going to arbitrarily reassign the reference. Could keep it private and construct one in New.

Reply:
done